config.yaml

```diff
@@ -3,28 +3,22 @@
 # See LICENSE file for licensing details.
 
 options:
-  zookeeper-properties:
-    description: zookeeper.properties configuration file.
+  additional-options:
+    description: "Newline-delimited list of additional custom options"
     type: string
-    default: |
-      clientPort=2181
-      dataDir=/var/snap/kafka/common/data
-      dataLogDir=/var/snap/kafka/common/log
-      dynamicConfigFile=/var/snap/kafka/common/zookeeper-dynamic.properties
-      tickTime=2000
-      initLimit=5
-      syncLimit=2
-      maxClientCnxns=60
-      minSessionTimeout=4000
-      maxSessionTimeout=40000
-      autopurge.snapRetainCount=3
-      autopurge.purgeInterval=0
-      reconfigEnabled=true
-      standaloneEnabled=false
-      4lw.commands.whitelist=*
-      DigestAuthenticationProvider.digestAlg=SHA3-256
-      quorum.auth.enableSasl=true
-      quorum.auth.learnerRequireSasl=true
-      quorum.auth.serverRequireSasl=true
-      authProvider.sasl=org.apache.zookeeper.server.auth.SASLAuthenticationProvider
-      audit.enable=true
+  client-port:
+    description: ""
+    type: int
+    default: 2181
```

Comment:
Do we need to expose this (or the secure port)? Feels like the sort of thing that could easily be solved with a reverse proxy. Note that these charms will mostly be deployed inside LXD containers or VMs where they are the only app running, so it's unlikely there will be conflict.

Reply:
I'm not super familiar with the use case, but what does providing the config hurt in this scenario?